.github/workflows/docker-release.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: docker release
+
+on:
+  release:
+    types: [released,prereleased,edited]
+
+env:
+  LATEST_RELEASE: 0.6
```

Comment:
Isn't it `1.0.0-rc2`? Can we automatically get this version from `pom.xml`?
Here is how we compute `commit` version of JanusGraph: 
https://github.com/JanusGraph/janusgraph/blob/029b4d2560c09cfc05393e7b7b9513dd92cc26da/.github/workflows/ci-publish-commit.yml#L61

I think we could do something similar to that. I.e. something similar like below (first get latest release, then shrink the string to the first 3 characters. I.e. "1.0" or "0.6"):
```
export LATEST_RELEASE="$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout | grep -Po \"^...\")"
```

Reply:
No,  It is not possible.

To follow our release guide:
1. Create tag
    * build image with e.g. janusgraph/janusgraph:v1.0.0-(hash) and push it
3. Vote for release
4. Mark release as ready
   * re tag with e.g. janusgraph/janusgraph:v1.0.0
   * If our released version is part of current main tag as janusgraph/janusgraph:latest. Why if we have have v1.1 and v1.0 want to fix something, it should re tag v1.0 as latest. 


Do you understand my description? Otherwise, we can call. We have to update the release process for it accordingly.

Reply:
Thank you @farodin91 for explanation. As far as I understand the purpose of this action is to re-tag `latest` tag to the highest version according to semantic versioning as well as to create proper version tags for already uploaded docker images. If so, having `LATEST_VERSION` here hardcoded is OK I think. Another solution could be to fetch all the tags and find the highest tag according to semantic versioning.
I'm good with `LATEST_VERSION` being hardcoded because finding latest tag according to semver might be more complicated.

Reply:
I opened a separate issue to automate latest version determining process, so that we could remove `LATEST_VERSION` in this file later: #3905

Reply:
I added this to https://github.com/JanusGraph/janusgraph/issues/3793